pkg/apis/serving/v1alpha1/cached_model_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2023 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageContainerSpec defines the container spec for the storage initializer init container, and the protocols it supports.
+// +k8s:openapi-gen=true
+type ClusterCachedModelSpec struct {
+	// Container spec for the storage initializer init container
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="StorageUri is immutable"
+	StorageUri string            `json:"storageUri" validate:"required"`
+	ModelSize  resource.Quantity `json:"modelSize" validate:"required"`
+	NodeGroups []string          `json:"nodeGroups" validate:"required"`
+	// only local is supported for now
+	StorageType   StorageType   `json:"storageType" validate:"required"`
+	CleanupPolicy CleanupPolicy `json:"cleanupPolicy" validate:"required"`
+}
```

Comment:
One missing field might be envvars for auth? I was thinking of storing envvars in a configmap but it's hard to refer to a secret in a config map.
```
    - name: AWS_ACCESS_KEY_ID
      valueFrom:
        secretKeyRef:
          key: access-key
          name: name
```